common/utils/cva.ts

```diff
@@ -36,4 +37,10 @@ export const cva: typeof cvaOriginal = (base, config) => {
   };
 };
 
-export type { VariantProps } from 'class-variance-authority';
+type ExcludeNull<T> = { [P in keyof T]: Exclude<T[P], null> };
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type VariantProps<Comp extends (...args: any) => any> = ExcludeNull<
+  VariantPropsOriginal<Comp>
+>;
+/* eslint-enable @typescript-eslint/no-explicit-any */
```

Comment:
oh! i see what you're doin. you just need: https://cva.style/docs/getting-started/typescript#extracting-variant-types:~:text=ads%20via%20Carbon-,Extracting%20Variant%20Types,-cva%20offers%20the

```
export const button = cva(/* … */);
export type ButtonProps = VariantProps<typeof button>;
```

right? or does our utility abstraction fn ruin it somehow?

---

if that doesn't work, would `Required<VariantProps<:COMP_IN_QUESTION:>>` not satisfy your needs?

Reply:
I did this because the current version of cva adds `null` and `undefined` to the variants. This will be fixed in the cva@beta according to this issue

https://github.com/joe-bell/cva/issues/253

Doing the required will need us to pass in the props and I think we would want to allow the defaults to take over.